compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1812,6 +1812,14 @@ rustc_queries! {
         desc { |tcx| "computing crate imported by `{}`", tcx.def_path_str(def_id) }
     }
 
+    /// Gets the number of definitions in this crate.
+    ///
+    /// This allows external tools to iterate over all definitions in a crate.
```

Comment:
```suggestion
    /// This allows external tools to iterate over all definitions in a foreign crate.
    ///
    /// This should never be used for the local crate, instead use XXX
```

Reply:
iter_local_def_ids or whatever it's called

Reply:
I'm also renaming this query to `num_extern_def_ids` to make this restriction more explicit.